### PR TITLE
Permit key-rotation of S3 encrypted objects

### LINKED ISSUE #7399: [SSE-S3] Key rotation in WORM mode
Allow SSE-S3 key rotation in WORM mode

## Expected Behavior

It should be possible to rotate keys for SSE-S3 encrypted objects, even if the server is started with WORM mode enabled. Since key rotation involves just mutating SSE-S3 specific metadata by re-wrapping the OEK (object encryption key) with a new KEK( key encryption key) derived from a new KMS data key and does not mutate the encrypted object itself, it would not violate the WORM guarantee.
 
## Current Behavior

Currently when WORM mode is enabled, we return a "The specified method is not allowed against this resource." error 

## Possible Solution

Allow key rotation operation in WORM mode
## Steps to Reproduce (for bugs)

1. MINIO_WORM=on ./minio server /tmp/data
2. aws s3api put-object --bucket tbucket11 --key q2   --no-verify-ssl --body /home/kris/data/qrd.csv --no-verify-ssl
3. aws s3api copy-object --bucket tbucket11 --copy-source tbucket11/q2 --key q2    --no-verify-ssl --endpoint-url http://localhost:9000 --server-side-encryption AES256 
4. Above operation causes a MethodNotAllowed error 

## Regression
 No

## Your Environment

* Version used (`minio version`):
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
* Link to your project:

### PR DESCRIPTION
**Pull request by poornas**
in WORM mode

Fixes : #7399

## Regression
 No

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#e86eb78c-d59a-4b68-b58d-eeaf38925b5b) |
<a name="e86eb78c-d59a-4b68-b58d-eeaf38925b5b"></a>
### 7429-f74a1d5/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:32148
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 6506e4668041:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 45 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 58 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 31 seconds
(8/14) Running minio-dotnet tests ... done in 50 seconds
(9/14) Running minio-go tests ... done in 54 seconds
(10/14) Running minio-java tests ... done in 2 minutes and 25 seconds
(11/14) Running minio-js tests ... done in 41 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 28 seconds
{
  "args": {
    "object_name": "5624cf97-ebd8-4cf2-ad03-3e99d2f1a404",
    "bucket_name": "minio-py-test-12f1e6fd-bb6f-4cc1-b990-0a2ecf09d8b0"
  },
  "duration": 100,
  "message": "AccessDenied: message: Access Denied",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "status": "FAIL",
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)"
}

Executed 11 out of 14 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#3601ca17-b348-49a3-be1b-e79971583bde) |
| mint-large-bucket.sh | :x: [more...](#ac0af180-3341-4997-83b0-97edb8e0d395) |
<a name="ac0af180-3341-4997-83b0-97edb8e0d395"></a>
### 7429-fa61529/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:30483
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 96c76eb53168:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 50 seconds
(4/14) Running aws-sdk-ruby tests ... done in 16 seconds
(5/14) Running awscli tests ... done in 2 minutes and 1 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... FAILED in 15 seconds
{
  "name": "mc",
  "duration": "5910",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-13066 --quiet --no-color cp myminio/mc-test-bucket-12428/mc-test-object-20882 mc-test-object-20882.downloaded  >>> \n`myminio/mc-test-bucket-12428/mc-test-object-20882` -> `mc-test-object-20882.downloaded`\nmc: <ERROR> Failed to copy `http://72.28.97.53:30483/mc-test-bucket-12428/mc-test-object-20882`. read tcp 172.17.0.9:35716->72.28.97.53:30483: read: connection reset by peer\nmc: <ERROR> Session safely terminated. To resume session `mc session resume jAdWcvPD`"
}

Executed 6 out of 14 tests successfully.
```
<a name="3601ca17-b348-49a3-be1b-e79971583bde"></a>
### 7429-fa61529/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:32283
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 92d7b7dfa1f8:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 45 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 54 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 22 seconds
(8/14) Running minio-dotnet tests ... done in 56 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 7 seconds
(10/14) Running minio-java tests ... done in 4 minutes and 5 seconds
(11/14) Running minio-js tests ... done in 43 seconds
(12/14) Running minio-py tests ... FAILED in 7 minutes and 28 seconds
{
  "args": {
    "bucket_name": "minio-py-test-2d6c4e94-6e22-4b14-80e8-d05c84b64765",
    "object_name": "58a41dcb-39ea-49cb-8df0-c8f3083461d1"
  },
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "duration": 89,
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "status": "FAIL",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "message": "AccessDenied: message: Access Denied"
}

Executed 11 out of 14 tests successfully.
```

**Review by aead — DISMISSED** [review]
LGTM

**Comment by aead** [comment]
CI reports error b/c of Go 11.4

**Comment by Praveenrajmani** [comment]
Can you check the travis failures @poornas

File: cmd/object-handlers.go [hunk 1]
```diff
@@ -720,16 +720,17 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	// Deny if WORM is enabled
-	if globalWORMEnabled {
+	cpSrcDstSame := isStringEqual(pathJoin(srcBucket, srcObject), pathJoin(dstBucket, dstObject))
+
+	// Deny if WORM is enabled. If operation is key rotation of SSE-S3 encrypted object
+	// allow the operation
+	if globalWORMEnabled && !(cpSrcDstSame && crypto.S3.IsRequested(r.Header)) {
```
Comment (Praveenrajmani) [inline cmd/object-handlers.go:727]: What if source and destination are same and the metadata directive is set to "REPLACE" with WORM enabled? Are we allowing it?

Comment (aead) [reply cmd/object-handlers.go:727]: I guess not - since that would be a "real" change to the object?!

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#09459ff7-1717-46b0-93c2-ba324f3ecd83) |
<a name="09459ff7-1717-46b0-93c2-ba324f3ecd83"></a>
### 7429-492a7c8/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:32188
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 486c210f9d5b:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 1 minutes and 7 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 2 minutes and 18 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 22 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 6 seconds
(9/14) Running minio-go tests ... done in 4 minutes and 0 seconds
(10/14) Running minio-java tests ... done in 14 minutes and 5 seconds
(11/14) Running minio-js tests ... done in 1 minutes and 10 seconds
(12/14) Running minio-py tests ... FAILED in 20 minutes and 20 seconds
{
  "message": "AccessDenied: message: Access Denied",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "status": "FAIL",
  "args": {
    "object_name": "697bb6bb-1f6f-425f-96d5-dfe76e503687",
    "bucket_name": "minio-py-test-3c2e430f-de44-41b1-9748-b23674cbcf5b"
  },
  "duration": 205,
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)"
}

Executed 11 out of 14 tests successfully.
```

**Comment by poornas** [comment]
@aead, can you reapprove? rebased PR for travis failure.

**Review by aead — DISMISSED** [review]

File: cmd/object-handlers.go (hunk 1 above)
Comment (Praveenrajmani) [reply cmd/object-handlers.go:727]: @poornas In the above case, the target metadata gets altered with the "REPLACE" option enabled. If you look here - https://github.com/minio/minio/pull/7429/files#diff-d1c4dccc78d8d9f128ca8972a1199176R949. Such requests should be responded with `ErrMethodNotAllowed`

Comment (poornas) [reply cmd/object-handlers.go:727]: @Praveenrajmani, you are right - updated PR with fix.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#2e320508-d326-4b57-890c-9a08b322ede1) |
<a name="2e320508-d326-4b57-890c-9a08b322ede1"></a>
### 7429-66f5595/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:32287
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9847aac5f6d2:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-lxh6howentz4gn",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 360022,
  "error": "XMinioWriteQuorum: Multiple disks failures, unable to write data.\n\tstatus code: 503, request id: 15931DA1EE825979, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Review by Praveenrajmani — APPROVED** [review]
LGTM

**Comment by aead** [comment]
SimpleCI is failing - apart from that LGTM

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :x: [more...](#baaf9603-ddea-4228-9227-32d3b14b9ff3) |
| mint-fs.sh | :x: [more...](#f28c9ff9-3474-44e9-958b-1533db6c97c6) |
| mint-dist-xl.sh | :x: [more...](#d5dd34c0-24ba-4fc3-bf36-1849235393d8) |
| mint-gateway-nas.sh | :x: [more...](#19b3c34b-3239-4eb2-95d4-a2bc7fe8a0a3) |
<a name="f28c9ff9-3474-44e9-958b-1533db6c97c6"></a>
### 7429-daa7a43/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:32317
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8a2f9e6745c6:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-m2nglogde6xdxk",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 2312,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:32317/aws-sdk-go-test-m2nglogde6xdxk: dial tcp 72.28.97.57:32317: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="19b3c34b-3239-4eb2-95d4-a2bc7fe8a0a3"></a>
### 7429-daa7a43/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:32738
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 26b1f86a6b48:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6e6oox16cqkcf3",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3073,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:32738/aws-sdk-go-test-6e6oox16cqkcf3: dial tcp 72.28.97.57:32738: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="d5dd34c0-24ba-4fc3-bf36-1849235393d8"></a>
### 7429-daa7a43/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:31702
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 6b485b84a0ab:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-fzaxl3ve6swby0",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 360010,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 1593DC6BAEF7553C, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="baaf9603-ddea-4228-9227-32d3b14b9ff3"></a>
### 7429-daa7a43/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:31904
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a07a712f22e5:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 30 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-vt1hhpe2ppcbra",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 390007,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.52:31904/aws-sdk-go-test-vt1hhpe2ppcbra: dial tcp 72.28.97.52:31904: i/o timeout",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Comment by poornas** [comment]
> SimpleCI is failing - apart from that LGTM

@aead, needs reapproval - SimpleCI failed to download ubuntu xenial

**Review by aead — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#4b48018b-f485-4a5e-9f6b-4238778626d6) |
| mint-dist-xl.sh | :x: [more...](#85ed42ad-3702-4fa8-bc0f-e11f89227343) |
<a name="85ed42ad-3702-4fa8-bc0f-e11f89227343"></a>
### 7429-223e161/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:32246
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8edf59a37ade:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 4 minutes and 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-ed1s2aat3cxutu",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 241171,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15942D977E35144C, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="4b48018b-f485-4a5e-9f6b-4238778626d6"></a>
### 7429-223e161/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:31044
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 24dadb39bf5a:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 46 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 53 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 27 seconds
(8/14) Running minio-dotnet tests ... done in 39 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 13 seconds
(10/14) Running minio-java tests ... done in 4 minutes and 35 seconds
(11/14) Running minio-js tests ... done in 46 seconds
(12/14) Running minio-py tests ... FAILED in 8 minutes and 57 seconds
{
  "duration": 165,
  "args": {
    "bucket_name": "minio-py-test-e813919c-2ad1-4444-9ea1-a56029c0a009",
    "object_name": "32fc1a2e-2714-4295-b92b-21954cc49cd9"
  },
  "status": "FAIL",
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "message": "AccessDenied: message: Access Denied"
}

Executed 11 out of 14 tests successfully.
```

**Pull request merged**